tests/tests_fabric/utilities/test_init.py

```diff
@@ -0,0 +1,40 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import mock
+
+import pytest
+import torch.nn
+
+from lightning.fabric.utilities.init import _EmptyInit
+from tests_fabric.helpers.runif import RunIf
+
+
+@mock.patch("lightning.fabric.utilities.init._TORCH_GREATER_EQUAL_1_13", False)
+def test_module_init_context_empty_weights_support():
+    with pytest.raises(NotImplementedError, match="Emtpy weight initialization requires PyTorch >= 1.13"), _EmptyInit():
+        pass
+
+
+# standalone because we need memory metrics isolated from other processes
+@RunIf(min_cuda_gpus=1, standalone=True, min_torch="1.13")
+def test_empty_init_memory_allocation():
+    """Test that no memory gets allocated when using the `_EmptyInit()` context manager."""
+    with _EmptyInit(enabled=True):
+        torch.nn.Linear(100, 100, device="cuda")
```

Comment:
I'm a bit suspicious of not assigning this to a variable, as it could get garbage collected as it is.